pysrc/bytewax/__init__.py

```diff
@@ -1,177 +1,13 @@
-        worker_count_per_proc: Number of worker threads to start on
-            each process.
-    """
-    addresses = _gen_addresses(proc_count)
-    with Pool(processes=proc_count) as pool:
-        futures = [
-            pool.apply_async(
-                cluster_main,
-                (
-                    flow,
-                    input_builder,
-                    output_builder,
-                    addresses,
-                    proc_id,
-                    worker_count_per_proc,
-                ),
-            )
-            for proc_id in range(proc_count)
-        ]
-        pool.close()
-
-        for future in futures:
-            # Will re-raise exceptions from subprocesses.
-            future.get()
-
-        pool.join()
-
-
-def run_cluster(
-    flow: Dataflow,
-    inp: Iterable[Tuple[int, Any]],
-    proc_count: int = 1,
-    worker_count_per_proc: int = 1,
-) -> List[Tuple[int, Any]]:
-    """Pass data through a dataflow running as a cluster of processes on
-    this machine.
-
-    Blocks until execution is complete.
-
-    Starts up cluster processes for you, handles connecting them
-    together, distributing input, and collecting output. You'd
-    commonly use this for notebook analysis that needs parallelism and
-    higher throughput, or simple stand-alone demo programs.
-
-    Input must be finite because it is reified into a list before
-    distribution to cluster and otherwise collected output will grow
-    unbounded.
-
-    See `spawn_cluster()` for starting a cluster on this machine with
-    full control over inputs and outputs.
-
-    See `cluster_main()` for starting one process in a cluster in a
-    distributed situation.
-
-    >>> flow = Dataflow()
-    >>> flow.map(str.upper)
-    >>> flow.capture()
-    >>> out = run_cluster(flow, [(0, "a"), (1, "b"), (2, "c")], proc_count=2)
-    >>> sorted(out)
-    [(0, 'A'), (1, 'B'), (2, 'C')]
-
-    Args:
-        flow: Dataflow to run.
-        inp: Input data. Will be reifyied to a list before sending to
-            processes. Will be partitioned between workers for you.
-        proc_count: Number of processes to start.
-        worker_count_per_proc: Number of worker threads to start on
-            each process.
-
-    Returns: List of `(epoch, item)` tuples seen by capture operators.
-    """
-    man = Manager()
-    inp = man.list(list(inp))
-
-    def input_builder(worker_index, worker_count):
-        for i, epoch_item in enumerate(inp):
-            if i % worker_count == worker_index:
-                yield epoch_item
-
-    out = man.list()
-
-    def output_builder(worker_index, worker_count):
-        return out.append
-
-    spawn_cluster(
-        flow, input_builder, output_builder, proc_count, worker_count_per_proc
-    )
-
-    return out
+from .bytewax import Dataflow, cluster_main, sleep_keep_gil, sleep_release_gil
+
+from .execution import run, spawn_cluster, run_cluster
+
+__all__ = [
```

Comment:
I'm not super familiar with this. Is there a specific behavior you're hoping to cause or prevent with `__all__`? Otherwise isn't importing a symbol into `__init__.py` enough to cause it to be included in `from bytewax import *`?